### PR TITLE
Create why_functional_programming.md

### PR DESCRIPTION
**Pull request by stephengoldbaum**

### CODE REVIEW

File: docs/why_functional_programming.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  total = 
+    suppliers
+      |> List.map .quantity
+      |> List.sum
+
+  max =
+    suppliers
+      |> List.maximumBy .quantity
+      |> List.map .supplierId
+
```
Comment (AttilaMihaly) [inline docs/why_functional_programming.md]: You should indent this one level deeper.

File: docs/why_functional_programming.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  in
+    (total, max)
```
Comment (AttilaMihaly) [inline docs/why_functional_programming.md]: This one should be indented to line up with the `in`.

Comment (AttilaMihaly) [reply docs/why_functional_programming.md]: Elm's standard is 4 spaces for indentation. I think that makes it easier to read.

**Comment by stephengoldbaum** [comment]
Publishing

**Pull request merged**